test/distributed/_composable/fsdp/test_fully_shard_training.py

```diff
@@ -300,9 +307,29 @@ def test_train_parity_multi_group_compile(self):
             self._test_train_parity_multi_group,
         )
 
+    @skip_if_lt_x_gpu(2)
+    def test_train_parity_multi_group_cpu_offload_eager(self):
+        """
+        Tests train parity against DDP when using multiple parameter groups for
+        communication and CPU offloading.
+        """
+        self.run_subtests(
+            {
+                "reshard_after_forward": [True],  # save CI time
+                "offload_policy": [CPUOffloadPolicy(True), CPUOffloadPolicy(False)],
```

Comment:
personally I misinterpreted ``CPUOffloadPolicy(False)`` as no cpu offloading, but actually it's ``pin_memory=False``. any chance this can be misinterpreated by others? 

flat-param FSDP use ``CPUOffload(False)`` to present ``offload_params=False``. this is probably why I was misinterpreting

Reply:
That is a good point. This difference might be confusing 🤔 

To explain some of my thoughts:
- For `mp_policy: MixedPrecisionPolicy`, since each of the dataclass's fields are optional, I did not want to have both `mp_policy=None` _and_ `mp_policy=MixedPrecisionPolicy(param_dtype=None, reduce_dtype=None)`, so I preferred just having `mp_policy=MixedPrecisionPolicy()` as default (no option for `None`).
- For `offload_policy`, now I am wondering whether it might be better to just have `offload_policy: Optional[OffloadPolicy] = None` (i.e. have it default to `None`). It does seem confusing as well that `OffloadPolicy` (base class) actually means no offloading.
- I am not sure how to make the fact that the `bool` arg to `CPUOffloadPolicy` is now changed to `pin_memory: bool` though.

Reply:
Can we not treat it as a kwarg where we use it in the code / documentation. Some style guides suggest only using Bools as kwarg for this reason to help avoid confusion.

Reply:
@Skylion007 Makes sense to me!